pkgs/applications/misc/ulauncher/default.nix

```diff
@@ -0,0 +1,70 @@
+{ stdenv, fetchurl, python27Packages,
+  gnome3, gobjectIntrospection, wrapGAppsHook, gtk3, webkitgtk,
+  libnotify, keybinder3, libappindicator, intltool, wmctrl }:
+
+python27Packages.buildPythonApplication rec  {
+  pname = "ulauncher";
+
+  versionNumber = "4.3.2";
+  version = "${versionNumber}.r8";
+
+  src = fetchurl {
+    url = "https://github.com/Ulauncher/Ulauncher/releases/download/${version}/ulauncher_${version}.tar.gz";
+    sha256 = "0494326s84nvyhblcghxzidp2qf0v0l61kyrf8a92qqryqnzmwip";
+  };
+
+  nativeBuildInputs = with python27Packages; [
+    dbus-python
+    distutils_extra
+    gobjectIntrospection
+    gnome3.defaultIconTheme
+    pygtk
+    intltool
```

Comment:
Only programs run during build should go to `nativeBuildInputs`, regular dependencies should go to `buildInputs` and python libraries to `propagatedBuildInputs`.

Reply:
I think most of the deps declaration I did at https://gist.github.com/worldofpeace/62626ac5cb437dc7e77b1ff7ee6a520f
was correct if it's still unclear.

Reply:
The non-python dependencies should not be propagated. Hopefully, in the future, python packages will not need this either.

Reply:
Is there anything else to be done before this thread is resolved?

Reply:
See https://github.com/NixOS/nixpkgs/pull/57859/files#r266724655